### DISCUSSION TITLE
Manually setting the position of the joystick

### DISCUSSION THREAD

**Discussion by basil-dsouza** [bobboteck/JoyStick#31] [Ideas]
This is a really handy project, and I appreciate the amount of work you have put into it.

Here is one function that I found useful to add to the source locally:
```
     /**
      * Resets the position of the joystick to center.
      * This is useful for joysticks that have autoReturnToCenter as false.
      */
     this.ResetPosition = function() 
     {
        movedX = centerX;
        movedY = centerY;
        // Delete canvas
        context.clearRect(0, 0, canvas.width, canvas.height);
        // Redraw object
        drawExternal();
        drawInternal();
     }

```

I guess techncially having a way to set x and y manually would also help. I might need that soon and if I do write it will share that with you as well.

**Comment by bobboteck** [comment]
It might be a cool feature, but how would you expect it to work?

Positioning the stick in the center is simple, but in other positions calculations must be made according to the size of the Canvas.

**Reply by basil-dsouza** [reply]
I havent tested this, but i reckon this should work:
    function setPosition(xPos, yPos) 
    {
       movedX = (xPos * maxMoveStick / 100.00 + centreX).toFixed();
       movedY = (yPos * -1 * maxMoveStick / 100.00 + centreY).toFixed();

       redrawJoystick(context, canvas);
    };

I also refactored some of the code to reduce the number of places the same code was called. Attached the overall file, Feel free to use it if you want. I was going to do a larger refactoring on it, but then ran out of time compared to the rest of the work i needed to do on the robot . (Couldnt attach the js, so converted it to txt and attached) [joy.txt](https://github.com/bobboteck/JoyStick/files/7849915/joy.txt)

**Reply by bobboteck** [reply]
Thanks, as soon as I have time I add the function and I'll test it.

I have configured Codacy code quality, and it suggest me to refactor the code every time 😄 , it's a work scheduled in my mind, tanks for your support on this task too.
